### PR TITLE
embed_wamr.md: improvements about threads

### PR DESCRIPTION
**Pull request by yamt**
- some more clarifications

- mention wasi-threads

- fix a confusing indentation

### CODE REVIEW

File: doc/embed_wamr.md [hunk 1]
```diff
@@ -258,68 +258,98 @@ We can't pass structure data or class objects through the pointer since the memo
...
-The `exec_env` is not thread safety, it will cause unexpected behavior if the same `exec_env` is used in multiple threads. However, we've provided two ways to execute wasm functions concurrently:
+It isn't safe to use an `exec_env` object in multiple threads concurrently.
+To run a multi-threaded application, you basically need a separate `exec_env`
+for each threads.
 
-- You can use `pthread` APIs in your wasm application, see [pthread library](./pthread_library.md) for more details.
+### Approaches to manage `exec_env` objects and threads
 
-- The `spawn exec_env` and `spawn thread` APIs are available, you can use these APIs to manage the threads in native:
+WAMR supports two approaches to manage `exec_env` and threads as described
```
Comment (loganek) [inline doc/embed_wamr.md:267]: Isn't wasi's spawn thread the third one?

Comment (yamt) [reply doc/embed_wamr.md:267]: "two" here means:

1. pthread (lib-pthread or wasi-threads)
2. wasm_runtime_spawn_exec_env/wasm_runtime_spawn_thread

File: doc/embed_wamr_spawn_api.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Thread related embedder API
+
+This document explains `wasm_runtime_spawn_exec_env` and
+`wasm_runtime_spawn_thread`.
+[Here](../samples/spawn-thread) is a sample to show how to use these APIs.
+
+  * spawn exec_env
+
+    `spawn exec_env` API creates a new `exec_env` based on the original `exec_env`. You use can use it in other threads. It's up to the embedder how to manage host threads to run the new `exec_env`.
```
Comment (wenyongh) [inline doc/embed_wamr_spawn_api.md:9]: `You use can use` -> `You can use`

File: doc/embed_wamr_spawn_api.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    ```C
+    new_exec_env = wasm_runtime_spawn_exec_env(exec_env);
+
+      /* Then you can use new_exec_env in your new thread */
+      module_inst = wasm_runtime_get_module_inst(new_exec_env);
+      func_inst = wasm_runtime_lookup_function(module_inst, ...);
+      wasm_runtime_call_wasm(new_exec_env, func_inst, ...);
```
Comment (wenyongh) [inline doc/embed_wamr_spawn_api.md:17]: Had better reduce the indent, make it same as above.

Comment (yamt) [reply doc/embed_wamr_spawn_api.md:17]: i guess this extra indentation means "this block might be executed in another thread".

File: doc/embed_wamr_spawn_api.md (hunk 2 above)
Comment (yamt) [reply doc/embed_wamr_spawn_api.md]: fixed. thank you.

**Review by wenyongh — COMMENTED** [review]
LGTM

File: doc/embed_wamr_spawn_api.md (hunk 3 above)
Comment (wenyongh) [reply doc/embed_wamr_spawn_api.md:17]: OK, it makes sense.

File: doc/embed_wamr.md (hunk 1 above)
Comment (loganek) [reply doc/embed_wamr.md:267]: I see. I wouldn't call `wasi-threads` a `pthread` - `pthread` is just one of the (possibly) many wrappers around `wasi-threads` (even though pthreads is currently the only implementation that wraps this hostcall). I was a bit confused by the line 273; given that `pthread` library is deprecated, we might point to `wasi-libc` instead

**Review by loganek — APPROVED** [review]
LGTM, left one suggestion

File: doc/embed_wamr.md (hunk 1 above)
Comment (yamt) [reply doc/embed_wamr.md:267]: ok. i tweaked wording a bit.

**Pull request merged**